src/Database/Expression/ComparisonExpression.php

```diff
@@ -315,3 +315,9 @@ protected function _collectExpressions($values): array
         return [$result, $expressions];
     }
 }
+
+// phpcs:disable
+// Comparison will not load during instanceof checks so ensure it's loaded here
+// @deprecated 4.1.0 Add backwards compatible alias.
```

Comment:
I am pretty sure IDEs and static analyzers won't warn about deprecated class usage unless the `@deprecated` tag is present in a docblock in `Comparion.php`.

Reply:
We never had them in class docblocks for the aliases. Does it warn if it's just in the file? What should I do?

Reply:
This could work [[source]](https://translate.google.com/translate?hl=en&sl=de&tl=en&u=https%3A%2F%2Fwww.schmengler-se.de%2F2016%2F09%2Fphp-class_alias-verwenden-um-klassen-rueckwaertskompatibel-zu-verschiebenumzubenennen%2F&prev=search):

```php
if (!class_exists(OldClass::class)) {
    /** @deprecated this is an alias for NewClass */
    class OldClass {}
}
```

Reply:
I feel like we could just add the deprecationWarning() to Comparison.php and the few users who manually create Comparison objects will get a warning.

Reply:
OK, fair enough I guess.

Reply:
That doesn't really work either apparently. If we create the alias in ComparisonExpression then Comparison.php is never loaded when users instantiate Comparison objects. They will only see the deprecation warning if they create them before running any cakephp code that creates ComparisonExpression objects.

Reply:
I tried to create a unit test, but it's unreliable, so I'm just adding the warning for that edge case.